x-pack/plugins/session_view/public/components/tty_player/hooks.ts

```diff
@@ -212,9 +232,27 @@ export const useXtermPlayer = ({
 
     return () => {
       window.removeEventListener('wheel', onScroll, true);
+      terminal.dispose();
     };
   }, [terminal, ref]);
 
+  const renderTruncatedMsg = useCallback(() => {
+    if (tty?.columns) {
+      const lineBreak = '-'.repeat(tty.columns);
+      const message = `${PROCESS_DATA_LIMIT_EXCEEDED} \x1b[1m${processName}.\x1b[22m ${PROCESS_DATA_LIMIT_EXCEEDED_2}`;
+      const link = policiesUrl
+        ? `\x1b[${Math.min(
+            message.length + 2,
+            tty.columns - VIEW_POLICIES.length - 4
+          )}G\x1b[1m\x1b]8;;${policiesUrl}\x1b\\[ ${VIEW_POLICIES} ]\x1b]8;;\x1b\\\x1b[22m`
+        : '';
+
+      return `\n\n\x1b[33m${lineBreak}\n${message}${link}\n${lineBreak}\x1b[0m
+
+`;
+    }
+  }, [policiesUrl, processName, tty?.columns]);
```

Comment:
should we let this function somewhere else i.e `tty_player/ansiHelpers.ts`? I think we will have more like this once we start working with alerts